omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/BaseResource1_12Test.java

```diff
@@ -0,0 +1,145 @@
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_12;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Assert;
+import org.junit.Before;
+import org.openmrs.Concept;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.HandlerExecutionChain;
+import org.springframework.web.servlet.mvc.annotation.AnnotationMethodHandlerAdapter;
+import org.springframework.web.servlet.mvc.annotation.DefaultAnnotationHandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+import static org.hamcrest.core.Is.is;
+
+public abstract class BaseResource1_12Test extends BaseModuleWebContextSensitiveTest {
```

Comment:
It seems this task is bigger than initially thought.
The idea was to move existing code from allergyapi with the least effort.
An initial assessment was the rest api aspect of allergyapi was not written in line with the way the current rest api was written.
Your review suggestions, indicate that this has to be completely revamped.
At the moment, I am porting the UI, allergyui, so it is difficult to see how a major rewrite here will impact the ui.